test/cli/fixtures/async-module-warning-promise.js

```diff
@@ -1,6 +1,24 @@
-QUnit.module('module manually returning a promise', function () {
-  QUnit.test('has a test', function (assert) {
-    assert.true(true);
+var errorFromModule;
+
+try {
```

Comment:
This updated version of the fixture is great for test/main/modules.js (which is where you wanted it originally). If we can make that one work with a then-able function, then I suggest restoring this like it was before as example of how it would fail "for real", and taking adventage of how TAP has a built-in ability to assert failures in-place. This has the benefit of both show-casing and verifying how it fails for real without trying to catch it at the same time.